enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,585 @@
+}
+
+// PolicyFulcioSubject defines the OIDC issuer and the email of the fulcio authentication configuration.
+type PolicyFulcioSubject struct {
+	// oidcIssuer contains the expected OIDC issuer.
+	// Example: "https://expected.OIDC.issuer/"
+	// Required if rootOfTrust.fulcioCAData is set.
+	// +kubebuilder:validation:Required
+	OIDCIssuer string `json:"oidcIssuer,omitempty"`
+
+	// signedEmail holds the email address the the certificate is issued for.
+	// Example: "expected-signing-user@example.com"
+	// Required if rootOfTrust.fulcioCAData is set.
+	// +kubebuilder:validation:Required
+	SignedEmail string `json:"signedEmail,omitempty"`
+}
+
+// PolicyIdentity defines image identity the signature claims about the image.
+// +union
+type PolicyIdentity struct {
+	// matchPolicy sets the type of matching to be used.
+	// Valid values are "MatchRepoDigestOrExact", "MatchRepository", "ExactRepository", "RemapIdentity". When omitted, the default value is "MatchRepoDigestOrExact".
+	// If set matchPolicy to ExactRepository, then the exactRepository must be specified.
+	// If set matchPolicy to RemapIdentity, then the remapIdentity must be specified.
+	// "MatchRepoDigestOrExact" means that the identity in the signature must be in the same repository as the image identity if the image identity is referenced by a digest. Otherwise, the identity in the signature must be the same as the image identity.
+	// "MatchRepository" means that the identity in the signature must be in the same repository as the image identity.
+	// "ExactRepository" means that the identity in the signature must be in the same repository as a specific identity specified by "repository".
+	// "RemapIdentity" means that the signature must be in the same as the remapped image identity. Remapped image identity is obtained by replacing the "prefix" with the specified “signedPrefix” if the the image identity matches the specified remapPrefix.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum=MatchRepoDigestOrExact;MatchRepository;ExactRepository;RemapIdentity
+	// +kubebuilder:validation:Required
+	MatchPolicy IdentityMatchPolicy `json:"matchPolicy,omitempty"`
+	// exactRepository is required if matchPolicy is set to "ExactRepository".
+	// +optional
+	PolicyMatchExactRepository *ExactRepository `json:"exactRepository,omitempty"`
+	// remapIdentity is required if matchPolicy is set to "RemapIdentity".
+	// +optional
+	PolicyMatchRemapIdentity *RemapIdentity `json:"remapIdentity,omitempty"`
+}
+
+// IdentityMatchPolicy defines the type of matching for setting "matchPolicy".
+// Valid values are "MatchRepoDigestOrExact", "MatchRepository", "ExactRepository", "RemapIdentity". When omitted, the default value is "MatchRepoDigestOrExact".
+type IdentityMatchPolicy string
+
+type PolicyMatchExactRepository struct {
+	// repository is the reference of the image identity to be matched.
+	// +kubebuilder:validation:Required
+	Repository string `json:"repository,omitempty"`
```

Comment:
What is the format of this? Any validation we can apply?

Reply:
I suggested we review the details about validation format and comment when we start working on openshfit/api after this enhancement is merged.

Reply:
Can you open the API PR in parallel then? IMO understanding the semantics and UX of the API is important and relates very directly to the enhancement process which is used to design our new interfaces and how users will interact with the feature

Reply:
@JoelSpeed The API PR is https://github.com/openshift/api/pull/1457